tests/unit/test_lda.py

```diff
@@ -0,0 +1,231 @@
+    with pytest.raises(ValueError):
+        LDA(num_topics=0, sagemaker_session=sagemaker_session, **COMMON_TRAIN_ARGS)
+
+
+def test_alpha0_validation_fail_type(sagemaker_session):
+    with pytest.raises(ValueError):
+        LDA(alpha0='other', sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_max_restarts_validation_fail_type(sagemaker_session):
+    with pytest.raises(ValueError):
+        LDA(max_restarts='other', sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_max_restarts_validation_fail_type2(sagemaker_session):
+    with pytest.raises(ValueError):
+        LDA(max_restarts=0.1, sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_max_restarts_validation_fail_value(sagemaker_session):
+    with pytest.raises(ValueError):
+        LDA(max_restarts=0, sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_max_iterations_validation_fail_type(sagemaker_session):
+    with pytest.raises(ValueError):
+        LDA(max_iterations='other', sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_max_iterations_validation_fail_type2(sagemaker_session):
+    with pytest.raises(ValueError):
+        LDA(max_iterations=1.1, sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_max_iterations_validation_fail_value(sagemaker_session):
+    with pytest.raises(ValueError):
+        LDA(max_iterations=0, sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_tol_validation_fail_type(sagemaker_session):
+    with pytest.raises(ValueError):
+        LDA(tol='other', sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_tol_validation_fail_value(sagemaker_session):
+    with pytest.raises(ValueError):
+        LDA(tol=0, sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+PREFIX = "prefix"
+BASE_TRAIN_CALL = {
+    'hyperparameters': {},
+    'image': registry(REGION, LDA.__name__) + '/lda:1',
+    'input_config': [{
+        'DataSource': {
+            'S3DataSource': {
+                'S3DataDistributionType': 'ShardedByS3Key',
+                'S3DataType': 'ManifestFile',
+                'S3Uri': 's3://{}/{}'.format(BUCKET_NAME, PREFIX)
+            }
+        },
+        'ChannelName': 'train'
+    }],
+    'input_mode': 'File',
+    'output_config': {'S3OutputPath': 's3://{}/'.format(BUCKET_NAME)},
+    'resource_config': {
+        'InstanceCount': TRAIN_INSTANCE_COUNT,
+        'InstanceType': TRAIN_INSTANCE_TYPE,
+        'VolumeSizeInGB': 30
+    },
+    'stop_condition': {'MaxRuntimeInSeconds': 86400}
+}
+
+FEATURE_DIM = 10
+MINI_BATCH_SZIE = 200
+HYPERPARAMS = {'num_topics': NUM_TOPICS, 'feature_dim': FEATURE_DIM, 'mini_batch_size': MINI_BATCH_SZIE}
+STRINGIFIED_HYPERPARAMS = dict([(x, str(y)) for x, y in HYPERPARAMS.items()])
+HP_TRAIN_CALL = dict(BASE_TRAIN_CALL)
+HP_TRAIN_CALL.update({'hyperparameters': STRINGIFIED_HYPERPARAMS})
+
+
+def test_call_fit(sagemaker_session):
```

Comment:
I don't think we should write unit tests that test functionality past the point where the LDA class hands off to the base class. The base class should already have test coverage on this functionality (and if it doesn't, we should add it there).

We should really just be (comprehensively) unit testing that the new code we wrote works correctly, and not more.

Reply:
I agree with that principle. These unit tests verify logic inside fit() on this level (verification of batch_size). This particular one tests that it delegates to base implementation correctly. It's easiest to check how it calls to the service since this is where we intercept calls.

Reply:
You should be able to use patching to replace the fit() method of the base class with a Mock object, and verify that parameters are passed correctly to the mock. That would be much more direct and concise (and thus maintainable).

Reply:
yeah - this seems cleaner, changing